catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventHandlerUtils.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.transformer.xml.streaming.lib;
+
+import static ddf.catalog.data.types.Validation.VALIDATION_WARNINGS;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.impl.AttributeImpl;
+
+public class SaxEventHandlerUtils {
+
+    public List<Attribute> getCombinedMultiValuedAttributes(Set<AttributeDescriptor> descriptors,
+            List<Attribute> attributes) {
+        Map<String, Boolean> multiValuedMap = getMultiValuedNameMap(descriptors);
+
+        List<String> validationWarnings = new ArrayList<>();
+        Map<String, Attribute> attributeMap = new HashMap<>();
+        for (Attribute attribute : attributes) {
+            String attributeName = attribute.getName();
+            Attribute addedAttribute = attributeMap.putIfAbsent(attributeName, attribute);
+
+            if (addedAttribute != null) {
+                Boolean addValue = multiValuedMap.get(attributeName);
```

Comment:
You construct one of these in the `impl` bundle and then inject it into all the handlers that need one, instead of newing one up in each handler. It doesn't look like this class uses state at all.